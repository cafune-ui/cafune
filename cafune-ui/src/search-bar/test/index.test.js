import SearchBar from '../';
import { render } from 'enzyme';
import { shallow, deep } from 'preact-render-spy';

describe('SearchBar', () => {
  it('should render with placeholder', () => {
    const wrapper = shallow(
      <SearchBar
        placeholder="请输入关键词"
        action={{
          type: 'cancel',
          onClick() {
            console.log('action cancel');
          }
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render with input event', () => {
    const onChange = jest.fn();
    const onConfirm = jest.fn();
    const onCancel = jest.fn();
    const val = 'test';
    const wrapper = shallow(
      <SearchBar
        value={val}
        onCancel={onCancel}
        onChange={onChange}
        onConfirm={onConfirm}
      />
    );
    expect(wrapper.find('input').attr('value')).toEqual(val);
    wrapper.find('input').simulate('change');
    expect(onChange).toHaveBeenCalled();
    wrapper.find('form').simulate('submit');
    expect(onConfirm).toHaveBeenCalled();
  });
  it('should call confirm', () => {
    const onConfirm = jest.fn();
    const wrapper = shallow(
      <SearchBar
        value="test"
        onConfirm={onConfirm}
        action={{
          type: 'confirm',
          onClick: onConfirm
        }}
      />
    );
    wrapper.find('Button').simulate('click');
    expect(onConfirm).toHaveBeenCalled();
  });
  it('should call cancel', () => {
    const onCancel = jest.fn();
    const wrapper = shallow(
      <SearchBar
        value="test"
        onCancel={onCancel}
        action={{
          type: 'cancel'
        }}
      />
    );
    wrapper.find('Button').simulate('click');
    expect(onCancel).toHaveBeenCalled();
  });
  it('should call clear', () => {
    const onCancel = jest.fn();
    const cancelClick = jest.fn();
    const wrapper = shallow(
      <SearchBar
        value="test"
        action={{
          type: 'cancel',
          onClick: cancelClick
        }}
        onChange={val =>
          wrapper.render(
            <SearchBar
              value={val}
              action={{
                type: 'cancel',
                onClick: cancelClick
              }}
            />
          )
        }
      />
    );
    wrapper.find('Button').simulate('click');
    expect(onCancel).not.toHaveBeenCalled();
    expect(cancelClick).toHaveBeenCalled();
    expect(wrapper.find('input').attr('value')).toBe('');
  });
});
