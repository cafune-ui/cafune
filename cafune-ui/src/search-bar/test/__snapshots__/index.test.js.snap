// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render with placeholder 1`] = `
RenderContext {
  "0": VNode {
    "attributes": Object {
      "action": Object {
        "onClick": [MockFunction],
        "type": "cancel",
      },
      "placeholder": "请输入关键词",
    },
    "children": Array [],
    "key": undefined,
    "nodeName": [Function],
  },
  "length": 1,
}
`;

exports[`SearchBar should render with placeholder 2`] = `
RenderContext {
  "0": VNode {
    "attributes": Object {
      "action": Object {
        "onClick": [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "type": "cancel",
      },
      "placeholder": "请输入关键词",
    },
    "children": Array [],
    "key": undefined,
    "nodeName": [Function],
  },
  "length": 1,
}
`;
